types/react-native/index.d.ts

```diff
@@ -3428,22 +3428,22 @@ export interface ViewabilityConfig {
 /**
  * @see https://facebook.github.io/react-native/docs/flatlist.html#props
  */
-export interface FlatListProperties<ItemT> {
+export interface FlatListProperties<ItemT> extends ScrollViewProperties {
 
     /**
      * Rendered in between each item, but not at the top or bottom
      */
     ItemSeparatorComponent?: React.ComponentClass<any> | null
 
     /**
-     * Rendered at the bottom of all the items.
+     * Rendered at the very end of the list.
      */
-    ListFooterComponent?: React.ComponentClass<any> | null
+    ListFooterComponent?: React.ComponentClass<any> | (() => React.ReactElement<any>) | null
```

Comment:
Yep, that's not quite right; you want to be able to provide a function that emits any element or you want to give it a component class that it can instantiate.  If you take a look at VirtualizedList.js, you can see how the prop is used to instantiate the component:

```if (ListHeaderComponent) {
      const element = React.isValidElement(ListHeaderComponent)
        ? ListHeaderComponent // $FlowFixMe
        : <ListHeaderComponent />;
      cells.push(
        <View key="$header" onLayout={this._onLayoutHeader}>
          {element}
        </View>
      );
    }```

This allows usage like the following: 

```
renderHeader = () => {
 return <View />
}```

and in the FlatList itself:
```
<FlatList
              data={this.props.data}
              renderItem={this.renderItem}
              keyExtractor={this.keyExtractor}
              ListHeaderComponent={this.renderHeader}
            />
```